tests/rest/client/test_models.py

```diff
@@ -11,14 +11,47 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import unittest
+import unittest as stdlib_unittest
 
-from pydantic import ValidationError
+from pydantic import BaseModel, ValidationError
 
-from synapse.rest.client.models import EmailRequestTokenBody
+from synapse.rest.client.models import EmailRequestTokenBody, ThreepidMedium
 
 
-class EmailRequestTokenBodyTestCase(unittest.TestCase):
+class ThreepidMediumEnumTestCase(stdlib_unittest.TestCase):
+    class Model(BaseModel):
+        medium: ThreepidMedium
+
+    def test_accepts_valid_medium_string(self) -> None:
```

Comment:
Wonder if we should roll our own `StrEnum` that also does `__repr__` and `__str__` in a less surprising way (i.e. by calling it on the string value)?
I suppose it ultimately doesn't matter for the sake of Pydantic

Reply:
Note that 3.11 will have a StrEnum in the stdlib: https://docs.python.org/3.11/library/enum.html?highlight=strenum#enum.StrEnum

There's a backport [here](https://pypi.org/project/backports.strenum/). If we do this I'd prefer to import from stdlib and fallback to the backport.

Reply:
absolutely fine with that idea, as long as we have confidence that the usual pitfall around `__str__` is addressed